From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Wed, 6 Mar 2024 11:21:07 +0000
Subject: [PATCH] fix arm asm meson source build

meson does mot support output with paths; add a meson.build file in the arm
directory. The output files were being incorrectly placed in the celt/ directory.

Program arm/arm2gnu.pl found: YES (/var/media/DATA/home-rudi/LibreELEC.kernel11/build.LibreELEC-H3.arm-12.0-devel/build/opus-v1.5.1/celt/arm/arm2gnu.pl)
Configuring celt_pitch_xcorr_arm-gnu.S with command

../celt/meson.build:51:25: ERROR: configure_file keyword argument "output" Output 'arm/armopts.s' must not contain a path segment.

before:
celt/celt_pitch_xcorr_arm-gnu.S

after:
celt/arm/celt_pitch_xcorr_arm-gnu.S
celt/arm/armopts.s
celt/arm/armopts-gnu.S

Signed-off-by: Rudi Heitbaum <rudi@heitbaum.com>
Signed-off-by: Jean-Marc Valin <jmvalin@jmvalin.ca>
---
 celt/arm/meson.build | 12 ++++++++++++
 celt/meson.build     |  9 +--------
 2 files changed, 13 insertions(+), 8 deletions(-)
 create mode 100644 celt/arm/meson.build

diff --git a/celt/arm/meson.build b/celt/arm/meson.build
new file mode 100644
index 000000000000..b4dec564fcbe
--- /dev/null
+++ b/celt/arm/meson.build
@@ -0,0 +1,12 @@
+arm2gnu = [find_program('arm2gnu.pl')] + arm2gnu_args
+celt_sources_arm_asm = configure_file(input: 'celt_pitch_xcorr_arm.s',
+  output: '@BASENAME@-gnu.S',
+  command: arm2gnu + ['@INPUT@'],
+  capture: true)
+celt_arm_armopts_s_in = configure_file(input: 'armopts.s.in',
+  output: 'armopts.s',
+  configuration: opus_conf)
+celt_arm_armopts_s = configure_file(input: [celt_arm_armopts_s_in],
+  output: '@BASENAME@-gnu.S',
+  command: arm2gnu + ['@INPUT@'],
+  capture: true)
diff --git a/celt/meson.build b/celt/meson.build
index 7852fc156021..46601ca13586 100644
--- a/celt/meson.build
+++ b/celt/meson.build
@@ -43,14 +43,7 @@ if host_cpu_family in ['arm', 'aarch64'] and have_arm_intrinsics_or_asm
     celt_sources += sources['CELT_SOURCES_ARM_NE10']
   endif
   if opus_arm_external_asm
-    arm2gnu = [find_program('arm/arm2gnu.pl')] + arm2gnu_args
-    celt_sources_arm_asm = configure_file(input: 'arm/celt_pitch_xcorr_arm.s',
-      output: '@BASENAME@-gnu.S',
-      command: arm2gnu + ['@INPUT@'],
-      capture: true)
-    celt_arm_armopts_s = configure_file(input: 'arm/armopts.s.in',
-      output: 'arm/armopts.s',
-      configuration: opus_conf)
+    subdir('arm')
     celt_static_libs += static_library('celt-armasm',
       celt_arm_armopts_s, celt_sources_arm_asm,
       install: false)
